### PR TITLE
[MAINTENANCE] Add support for `Validator` in Checkpoint CRUD

### PR DESCRIPTION
**Pull request by cdkini**
Like `SimpleCheckpoint`, allow the user to pass in a `Validator` instead of an input `validations` list.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -1981,6 +1993,11 @@ def add_or_update_checkpoint(
 
     @public_api
     @new_method_or_class(version="0.15.48")
+    @new_argument(
```
Comment (NathanFarmer) [inline great_expectations/data_context/data_context/abstract_data_context.py:1996]: ❤️

File: great_expectations/data_context/types/base.py [hunk 2]
```diff
@@ -2586,7 +2586,17 @@ def __str__(self) -> str:
 
 
 class CheckpointValidationConfig(AbstractConfig):
-    def __init__(self, id: Optional[str] = None, **kwargs: dict) -> None:
+    def __init__(
+        self,
+        id: str | None = None,
+        expectation_suite_name: str | None = None,
+        expectation_suite_ge_cloud_id: str | None = None,
+        batch_request: BatchRequestBase | dict | None = None,
```
Comment (NathanFarmer) [inline great_expectations/data_context/types/base.py]: So this doesn't accept Fluent `BatchRequest`?

Comment (Kilo59) [reply great_expectations/data_context/types/base.py]: 😆 I think we are going to be talking about this today in mob-typing.
But yeah I think this needs to be a union of `BatchRequestBase` and the fluent BatchRequest.

File: great_expectations/checkpoint/checkpoint.py [hunk 3]
```diff
@@ -960,9 +962,17 @@ def construct_from_config_args(
...
+        validator: Validator | None = None,
     ) -> Checkpoint:
         checkpoint_config: Union[CheckpointConfig, dict]
 
+        if validator:
+            if validations:
+                raise ValueError(
+                    "Please provide either a validator or validations list (but not both)."
+                )
+            validations = validator.convert_to_checkpoint_validations_list()
```
Comment (Kilo59) [inline great_expectations/checkpoint/checkpoint.py]: I assume defining overloads for this method is a huge pain???

**Review by Kilo59 — APPROVED** [review]
Could we add a test that using a checkpoint with a validator to `tests/datasource/fluent/test_viral_snippet.py`?

File: great_expectations/checkpoint/checkpoint.py (hunk 3 above)
Comment (cdkini) [reply great_expectations/checkpoint/checkpoint.py]: yeah I tried for a bit and decided to defer until that Checkpoint epic

File: great_expectations/data_context/types/base.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/data_context/types/base.py]: I'll add it!

**Pull request merged**